src/erl/clj_emitter.erl

```diff
@@ -1711,7 +1711,9 @@ clause({PatternExpr, BodyExpr}, StateAcc) ->
   {GuardAst0, StateAcc2}     = pop_ast(ast(GuardExpr, StateAcc1)),
   {BodyAst0, StateAcc3}      = pop_ast(ast(BodyExpr, StateAcc2)),
 
-  {[PatternAst1], PatGuards} = clj_emitter_pattern:pattern_list([PatternAst0]),
+  { [PatternAst1]
+  , PatGuards
+  } = clj_emitter_pattern:patterns([PatternAst0], locals(EnvPattern)),
```

Comment:
We only want to take into account the locals with the same name for case and receive. Other forms that create bindings between symbols and values can override the existing binding (e.g. `let` and `loop`).